prog/dftb+/lib_dftbplus/main.F90

```diff
@@ -1063,7 +1063,7 @@ subroutine processGeometry(this, env, iGeoStep, iLatGeoStep, tWriteRestart, tSto
             & this%extPressure, this%cellVol, this%dftbEnergy(1)%TS, this%tAtomicEnergy,&
             & this%dispersion, this%tPeriodic, this%tSccCalc, this%invLatVec, this%kPoint,&
             & this%iAtInCentralRegion, this%electronicSolver, this%reks,&
-            & allocated(this%thirdOrd), this%isRangeSep)
+            & allocated(this%thirdOrd), this%isRangeSep, qNetAtom=this%qNetAtom)
```

Comment:
@bhourahine In this commit, I add printing of qNetAtom to detailed.out file. In this block (1041 ~ 1070), why was this part added? Is both blocks (~1060 and (~790 or ~1020)) needed?

Reply:
@inseonglee if I understand the question correctly, this is a consequence of updating the self-consistent dispersion during the SCC loop, but not updating it for non-self-consistent dispersion until after the loop terminates (this leads to a more complicated logic to print the detailed.out data but is required as polarizability can only be evaluated from the output charges after at least one diagonalization).
Probably after the next release, this will be refactored to only print total energies after the loop terminates, in part to also refactor the electrostatic potential evaluation to only happen once in each cycle of the loop.This means the stationary, not the variational, SCC-DFTB energy would be available during the SCC loop.
The disadvantage of these changes is that in an aborted calculation, there will be no indication of the energy. It would only be once the SCC loop correctly terminates that all components of the total energy are available. The advantages are a somewhat faster calculation and a simpler code structure.

Reply:
OK, I see. Then, this part will be refactored after next release. I will not change this part.